### ISSUE TITLE
tooltips and A11Y for content type indicators

### ISSUE THREAD

**Issue by indirectlylit** [learningequality/kolibri#1979]
![image](https://user-images.githubusercontent.com/2367265/29224484-6f7caf3c-7e7f-11e7-9f6e-e7a40252a1d9.png)

The content type icons should have tooltips on hover and a11y labels for screen readers

**Comment by indirectlylit** [comment]
see also https://github.com/learningequality/kolibri/pull/2062 - the new keen tooltips might include a11y

**Comment by indirectlylit** [comment]
Update: I've worked on this a bit and hit some complexities:

* Keen's `ui-tooltip` component seems a bit buggy, and doesn't always work. In particular, when you pass a `ref` name as the `trigger` prop, the ref is sometimes not found and so the tooltip doesn't initialize properly
* I was able to get it to work when attaching the tooltip directly to the inline SVG in `content-icon`, but not at any of the higher-level DOM nodes. It really ought to be attached to `card-content-icon-wrapper` inside the `content-card` component.
* I've looked at a number of alternative tooltip libs, but all have disadvantages

I'm going to table this for now - might come back to it later.

**Comment by christianmemije** [comment]
@indirectlylit There was a fix related to `UiTooltip` in 1.0.1 Not sure if that might help https://github.com/JosephusPaye/Keen-UI/releases/tag/v1.0.1

**Comment by indirectlylit** [comment]
Yup - thanks. It does seem related but does not address the issues that I'm seeing

**Comment by christianmemije** [comment]
Got it. Did you take a look at [popperjs](https://github.com/FezVrasta/popper.js)?

**Comment by indirectlylit** [comment]
I tried this vue wrapper around popper: https://www.npmjs.com/package/vue-directive-tooltip

Pro:

* It was easy to set up and worked as expected.

Cons:

* The show/hide behavior was finicky
  * It would sometimes hide while I was still hovering on the element
  * It would sometimes stay visible even after moving to a new popup target
* The styling wasn't pretty out-of-the-box

It's possible that the negatives can be avoided, but I don't want to spend any more time on this for a bit

**Comment by indirectlylit** [comment]
might want to take another look at this now, because the content icon background changed implementation recently

**Comment by christianmemije** [comment]
@indirectlylit We could add a `showTooltip` prop to the `k-icon` component #2435
